src/features/posts/postsSlice.ts

```diff
@@ -0,0 +1,53 @@
+/* eslint no-param-reassign: "error" */
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Post } from '../../types/Post';
+import { getUserPosts } from '../../api/posts';
+
+type PostsState = {
+  posts: Post[],
+  loaded: boolean,
+  hasError: boolean,
+};
+const initialState: PostsState = {
+  posts: [],
+  loaded: true,
+  hasError: false,
+};
+
+export const getAsyncPosts = createAsyncThunk(
+  'posts/get',
+  async (userId: number) => {
+    const posts = await getUserPosts(userId);
+
+    return posts;
+  },
+);
+
+const postsSlice = createSlice({
+  name: 'posts',
+  initialState,
+  reducers: {
+    reset: (state) => {
+      state.posts = [];
+    },
```

Comment:
When resetting the state, also reset the `loaded` and `hasError` properties to their initial values.